anvil/src/eth/backend/mem/mod.rs

```diff
@@ -407,6 +407,37 @@ impl Backend {
         self.db.write().revert(id)
     }
 
+    /// Write all chain data to serialized bytes buffer
+    pub fn dump_state(&self) -> Result<Bytes, BlockchainError> {
+        self.db
+            .read()
+            .dump_state()
+            .map(|s| bytes::Bytes::from(serde_json::to_string(&s).unwrap_or_default()).into())
+            .ok_or_else(|| {
+                RpcError::invalid_params(
+                    "Dumping state not supported with the current configuration",
+                )
+                .into()
+            })
+    }
+
+    /// Deserialize and add all chain data to the backend storage
+    pub fn load_state(&self, buf: Bytes) -> Result<bool, BlockchainError> {
+        let state: SerializableState = serde_json::from_str(
+            str::from_utf8(&buf.0).map_err(|_| BlockchainError::FailedToDecodeStateDump)?,
```

Comment:
`serde_json::from_slice()` should be able to deserialize this directly

Reply:
2259dd8